File: YoutubeRadioApplication/src/main/java/com/ldz/project/model/UserRegister.java
package com.ldz.project.model;

import org.hibernate.validator.constraints.Email;
import org.springframework.stereotype.Component;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

/**
 * Created by ldalzotto on 15/04/2017.
 */
@Component
public class UserRegister {

    @NotNull
    @Size(min = 4)
    private String username;

    @NotNull
    @Size(min = 4)
    private String password;

    @NotNull
    @Email
    private String email;

    @NotNull
    private String ipaddress;

    @NotNull
    private String country;

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getIpaddress() {
        return ipaddress;
    }

    public void setIpaddress(String ipaddress) {
        this.ipaddress = ipaddress;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: YoutubeRadioApplication/src/scala/com/ldz/project/model/UserRegister.scala
package com.ldz.project.model

import javax.validation.constraints.{NotNull, Size}

import org.hibernate.validator.constraints.Email

import scala.annotation.meta.field
import scala.beans.BeanProperty

/**
  * Created by ldalzotto on 23/06/2017.
  */
case class UserRegister(@BeanProperty @(NotNull@field) @(Size@field)(min = 4) var username: String,
                        @BeanProperty @(NotNull@field) @(Size@field)(min = 4) var password: String,
                        @BeanProperty @(NotNull@field) @(Email@field) var email: String,
                        @BeanProperty @(NotNull@field) var ipaddress: String,
                        @BeanProperty @(NotNull@field) var country: String) {


  def this() {
    this(null, null, null, null, null)
  }

}
